nix/ocaml.nix

```diff
@@ -99,34 +99,7 @@ let
             --set MINA_BRANCH "''${MINA_BRANCH-<unknown due to nix build>}"
         done
       '') package.outputs);
-
-      runMinaCheck = { name ? "check", extraInputs ? [ ], extraArgs ? { } }:
-        check:
-        self.mina-dev.overrideAttrs (oa:
-          {
-            pname = "mina-${name}";
-            buildInputs = oa.buildInputs ++ extraInputs;
-            buildPhase = check;
-            outputs = [ "out" ];
-            installPhase = "touch $out";
-          } // extraArgs);
-    in {
-      # https://github.com/Drup/ocaml-lmdb/issues/41
-      lmdb = super.lmdb.overrideAttrs
-        (oa: { buildInputs = oa.buildInputs ++ [ self.conf-pkg-config ]; });
-
-      sodium = super.sodium.overrideAttrs (_: {
-        NIX_CFLAGS_COMPILE = "-I${pkgs.sodium-static.dev}/include";
-        propagatedBuildInputs = [ pkgs.sodium-static ];
-        preBuild = ''
-          export LD_LIBRARY_PATH="${super.ctypes}/lib/ocaml/${super.ocaml.version}/site-lib/ctypes";
-        '';
-      });
-
-      rpc_parallel = super.rpc_parallel.overrideAttrs
-        (oa: { buildInputs = oa.buildInputs ++ [ self.ctypes ]; });
-
-      mina-dev = pkgs.stdenv.mkDerivation ({
+      minaImpl = profile: pkgs.stdenv.mkDerivation ({
         pname = "mina";
         version = "dev";
```

Comment:
should this match `profile`? 🤔

Reply:
What do you mean by "match"?

P.S. probably I'll take out nix changes to a separate PR

Reply:
I meant `version = profile;` I'm not sure what the implications are of having the versions match, or where (if at all) this version is used...

Reply:
Good point. I'll investigate it tomorrow. This one I've put just the most straightforward way I could imagine to test it working and unblock Edmund

Reply:
I re-implemented this change in a more nix-conventional way in #15026